pkg/asset/rhcos/image.go

```diff
@@ -52,101 +55,107 @@ func (i *Image) Dependencies() []asset.Asset {
 func (i *Image) Generate(ctx context.Context, p asset.Parents) error {
 	if oi, ok := os.LookupEnv("OPENSHIFT_INSTALL_OS_IMAGE_OVERRIDE"); ok && oi != "" {
 		logrus.Warn("Found override for OS Image. Please be warned, this is not advised")
-		*i = Image(oi)
+		*i = *MakeAsset(oi)
 		return nil
 	}
 
 	ic := &installconfig.InstallConfig{}
 	p.Get(ic)
 	config := ic.Config
-	osimage, err := osImage(ctx, config)
+	osimageControlPlane, osImageCompute, err := osImage(ctx, config)
 	if err != nil {
 		return err
 	}
-	*i = Image(osimage)
+	*i = Image{osimageControlPlane, osImageCompute}
 	return nil
 }
 
+// For compatibility with before the asset had multi-arch support, returns the same image for both controlPlane and
+// compute.
+//
 //nolint:gocyclo
-func osImage(ctx context.Context, config *types.InstallConfig) (string, error) {
+func osImage(ctx context.Context, config *types.InstallConfig) (string, string, error) {
 	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
 	defer cancel()
 
 	archName := arch.RpmArch(string(config.ControlPlane.Architecture))
```

Comment:
this is still using the cp arch...ideally it should loop through the compute arches and fetch the images?

Reply:
Because that still has to function for cloud platforms that don't support multi-arch. Why is it a problem here?

Reply:
Changed according to another suggestion of yours.